### DISCUSSION TITLE
How does the NewPipe optimize the smoothness of audio playback?

### DISCUSSION THREAD

**Discussion by leo9082** [TeamNewPipe/NewPipe#11408] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

In my app, I use NewPipeExtractor to get the youtube video address to play.

But I found that the playback often freezes. The reason is that the audio playback often freezes.

I found NewPipe is very smooth. Can you tell me how NewPipe does it? How to optimize the audio stuttering problem.

**Comment by SamSirry** [comment]
I apologize for the cliché answer, but you have the source code wide open. And possibly ChatGPT has already seen it.

I don't know how you're testing your app, but it sounds like the issue could be related to audio drivers. I recommend testing on different brands of devices.
